src/Azure/Orleans.Clustering.AzureStorage/OrleansSiloInstanceManager.cs

```diff
@@ -114,17 +114,10 @@ private static Uri ConvertToGatewayUri(SiloInstanceTableEntry gateway)
         public async Task<IList<Uri>> FindAllGatewayProxyEndpoints()
         {
             if (logger.IsEnabled(LogLevel.Debug)) logger.Debug(ErrorCode.Runtime_Error_100277, "Searching for active gateway silos for deployment {0}.", this.DeploymentId);
-            const string zeroPort = "0";
 
             try
             {
-                string filterOnPartitionKey = TableQuery.GenerateFilterCondition(nameof(SiloInstanceTableEntry.PartitionKey), QueryComparisons.Equal,
-                    this.DeploymentId);
-                string filterOnStatus = TableQuery.GenerateFilterCondition(nameof(SiloInstanceTableEntry.Status), QueryComparisons.Equal,
-                    INSTANCE_STATUS_ACTIVE);
-                string filterOnProxyPort = TableQuery.GenerateFilterCondition(nameof(SiloInstanceTableEntry.ProxyPort), QueryComparisons.NotEqual, zeroPort);
-                string query = TableQuery.CombineFilters(filterOnPartitionKey, TableOperators.And, TableQuery.CombineFilters(filterOnStatus, TableOperators.And, filterOnProxyPort));
-                var queryResults = await storage.ReadTableEntriesAndEtagsAsync(query);
+                var queryResults = await storage.ReadTableEntriesAndEtagsAsync($"PartitionKey eq '{DeploymentId}' and Status eq 'Active' and ProxyPort ne '0'");
```

Comment:
FYI - you can use `TableOdataFilter.Create` to handle odata query escaping (note that no single quoting is required). And for special types like Guid or values containing single quotes, the escaping syntax is not intuitive.
```suggestion
                const string ACTIVE = "Active";
                const string ZERO= "0";
                var queryResults = await storage.ReadTableEntriesAndEtagsAsync(TableOdataFilter.Create($"PartitionKey eq {DeploymentId} and Status eq {ACTIVE} and ProxyPort ne {ZERO}"));
```